str/runners/str_iterative_eh_runner.py

```diff
@@ -49,45 +46,56 @@ def main(
     )
     b = hb.Batch(backend=backend, default_image=os.getenv('DRIVER_IMAGE'))
 
-    external_id_to_cpg_id: dict[str, str] = SampleApi().get_sample_id_map_by_external(
-        dataset, list(external_wgs_ids)
-    )
-    cpg_id_to_external_id = {
-        cpg_id: external_wgs_id
-        for external_wgs_id, cpg_id in external_id_to_cpg_id.items()
-    }
     if dataset in ['tob-wgs', 'hgdp']:
         ref_fasta = (
             'gs://cpg-common-main/references/hg38/v0/Homo_sapiens_assembly38.fasta'
         )
     else:
         ref_fasta = reference_path('broad/ref_fasta')
-    if dataset == 'tob-wgs':
-        analysis_query_model = AnalysisQueryModel(
-            sample_ids=list(external_id_to_cpg_id.values()),
-            projects=[dataset],
-            type=AnalysisType('cram'),
-            status=AnalysisStatus('completed'),
-            meta={'sequencing_type': 'genome', 'source': 'nagim'},
-        )
-    else:
-        analysis_query_model = AnalysisQueryModel(
-            sample_ids=list(external_id_to_cpg_id.values()),
-            projects=[dataset],
-            type=AnalysisType('cram'),
-            status=AnalysisStatus('completed'),
-            meta={'sequencing_type': 'genome'},
-        )
-    crams_path = AnalysisApi().query_analyses(analysis_query_model)
-    cpg_sids_with_crams = set(sid for sids in crams_path for sid in sids['sample_ids'])
-    cpg_sids_without_crams = set(cpg_id_to_external_id.keys()) - cpg_sids_with_crams
-    if cpg_sids_without_crams:
-        external_wgs_sids_without_crams = ', '.join(
-            cpg_id_to_external_id[sid] for sid in cpg_sids_without_crams
-        )
+
+    cram_retrieval_query = gql(
+        """
+        query MyQuery($dataset: String!,$input_cpg_sids: [String!]!) {
+    project(name: $dataset) {
+        sequencingGroups(id: {in_: $input_cpg_sids}) {
+        id
+        sample {
+            externalId
+        }
+        analyses(type: {eq: 'cram'}, active: {eq: true}) {
+            output
+            timestampCompleted
+        }
+        }
+    }
+    }
+        """
+    )
+    response = query(
+        cram_retrieval_query,
+        variables={'dataset': dataset, 'input_cpg_sids': input_cpg_sids},
+    )
+    valid_cpg_ids = []
+    crams_path = []
+    for i in response['project']['sequencingGroups']:
+        valid_cpg_ids.append(i['id'])
+        if (
+            len(i['analyses']) > 1
+        ):  # if there are multiple crams stored per CPG sample ID
```

Comment:
should there be an `else` here? It looks like you're not storing anything for SGs with only one CRAM?

Reply:
LOL

Reply:
yes good pick up